x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -177,18 +175,17 @@ private Tuple<ResolvedEnrichPolicy, String> mergeLookupResults(
             return Tuple.tuple(null, reason);
         }
         Map<String, EsField> mappings = new HashMap<>();
-        Collection<String> enrichFields = new LinkedHashSet<>();
         Map<String, String> concreteIndices = new HashMap<>();
         ResolvedEnrichPolicy last = null;
         for (Map.Entry<String, ResolvedEnrichPolicy> e : policies.entrySet()) {
             ResolvedEnrichPolicy curr = e.getValue();
             if (last != null && last.matchField().equals(curr.matchField()) == false) {
-                String error = "enrich policy [" + policyName + "] has different match_field across clusters";
-                return Tuple.tuple(null, error);
+                String detailed = "[" + last.matchField() + "] vs [" + curr.matchField() + "]";
```

Comment:
Not a big deal, more a consistency of style: in ql the error messages are usually of the form "this should have happened, I was expecting [ ], but I got [ ]".

For this particular scenario, I would have chosen something like:
`"enrich policy [" + policyName + "] has different match_field across clusters; expected [" + last.matchField() + "] but got [" + curr.matchField() + "]"`.

Same goes for the rest of the error messages in this resolver.

Reply:
++ I've revised these error messages. They are slightly different from your suggestion. `enrich policy [author] has different match types [match, range] across clusters`.